### ISSUE TITLE
Request: improve readability of oracle list

### ISSUE THREAD

**Issue by rsek** [XenotropicDev/TheOracle#19]
at this point i often resort to copy-pasting the !oraclelist output to a text editor to find the entry i'm looking for. i'm not sure what the solution here is, honestly; it's a lot of text in the first place because ironsworn/starforged has a lot of oracles, and there's only so much that can be done to compress that.

one way to approach it might be: accept that it's going to be long no matter what, and just make it as readable as possible even if it takes up more space to do so (and it almost certainly will). as it is, the post already takes up about half the vertical space on my 15" laptop screen, and it'll probably only expand as i get around to adding better aliases and as starforged adds more oracles. maybe automatically populate the "delete post" ❌ emoji react to remind people that they can dismiss the humongous post when they're done looking at it.

**Comment by XenotropicDev** [comment]
I'm not quite sure what can be done for this, but it's something that could be looked at. Currently it filters out the other tables if a default game is set, which helps cut down some, but there's still a lot. If someone has some ideas on what can be done to improve the look/organization I'm open to them.

**Comment by rsek** [comment]
so this occurred to me as i was playing with preformatted text tables for theme/domain reference output, and tried putting them in to embed fields:

<img width="532" alt="Screen Shot 2020-11-11 at 6 40 22 AM" src="https://user-images.githubusercontent.com/5354757/98824956-c803a900-23e8-11eb-9a03-380d974419c6.png">

which i think is a start, and probably more readable than the big wall o' text, but only just.

but check out the restrictions: maximum of 25 fields per post, 1024 characters per field. with that much space, we could combine multiple commands into a single field. add some bold/underline (you can wrap bold/underline around preformatted text to, er, format it, and bullets so there's a visual hierarchy... and with enough line breaks it creates a crude yet fairly readable column layout! also means we could give them some category headers if we wanted (alphabetic maybe? or maybe just some decorative emoji for fun, cuz why not):

<img width="536" alt="Screen Shot 2020-11-11 at 6 56 15 AM" src="https://user-images.githubusercontent.com/5354757/98826804-01d5af00-23eb-11eb-9df2-ae35556ad616.png">

it'd even play nice with mobile, probably.

i haven't done it with the complete data yet but but from the look of it, we could actually fit all the oracles in to a single post like this and have it remain legible.

**Comment by XenotropicDev** [comment]
The issue I foresee with this (not that the current implementation isn't guilty of this too) is that this will be really hard on mobile, because there are no inline fields for mobile. But overall I think this looks way better.

**Comment by rsek** [comment]
so it'd end up a bunch of columns stacked on top of one another? hmm.

it might be worth testing that output on mobile anyways to see if it's onerous; the scrolling touch-gesture feels pretty natural and fluid to most people, so mobile UX can often tolerate way more height than than a desktop/laptop screen. put another way, few people bat an eye at casually scrolling through a list of items in a single long column on mobile, but on desktop that doesn't feel as good.

i'd also be interested to see how discord's viewport moves on mobile when a new post exceeds the height of the viewport. like, does the viewport focus the top of the last message, meaning the user scrolls down to see the rest, or does it focus the end of the message, meaning the user must scroll up to see the rest? is it different when the window is focused or not?

all this is speculation, though. i think to make an informed decision the best course of action is to create test output and see how people feel about it.

in order to reduce the amount of scrolling, i could drop it down to two inline columns, and put more information before each line break. that is, given command and its aliases would be a comma-seperated list on a single line, perhaps with the primary command bolded at the start of the line. and there would be less wasted horizontal space on mobile with wider columns like that. (i'll probably do a mockup of this shortly, and see if i can simulate the single column layout whether with some mobile screenshots or by some other means).

another possible approach might be an emoji react to toggle between two different formats of post. like, a mobile-oriented version with some other format, and a desktop-oriented one? not as elegant as a (the discord approximation of) a responsive UX that automatically changes to suit the viewport size, and potentially impractical, but this is another thing where testing is the most efficient way to figure out whether it works or not (though only after simpler solutions are explored, probably)

oh, and, this is probably a nitpick, but... would it be possible to get the command output in all lower case? rationale: in the context of the styling/bullets that might be used to create a visual hierarchy, or other context cues like including the entire oracle command, __**`Action`**__ is a little more ambiguous as to whether it's a header or a command or both (especially in the context of discord embeds, where the backtick that normally might be used for other purposes), while __**`action`**__ is more likely to be read as a command because it's not following the typical grammatical rule of a header being in title case. (part of this ambiguity is probably my own fault for introducing the dang bullets, but it might be a good convention to follow in all documentation).

**Comment by XenotropicDev** [comment]
Overall I like the multi-column format you prototyped, so I'll try to emulate that. At worst the mobile layout is a wash with the current format, but the desktop experience will be much better

**Comment by rsek** [comment]
i did a couple variations using the starforged oracles, which handle a bit differently because some of their names get a bit long; they might be approaches to consider on a mobile view. also, they have proper categories that can be use as field titles. if we want to keep everything nicely lined up in columns, we could divide them among three columns and 'fake' the field names/category headers by emulating their styling.

the most universally useful idea in these might be the note included at the top of each category, suggesting the user invoke the command for an interactive post if they want all the rolls done at once (wording could be better).

the starforged oracles seem less tolerant of the narrower columns. also, the varying heights of each category leave a bunch of empty space:

<img width="511" alt="Screen Shot 2020-11-11 at 11 59 15 PM" src="https://user-images.githubusercontent.com/5354757/98911728-ec5b9600-2479-11eb-848e-1a1387497ad4.png">

mostly single-column layout with only the first two set to inline:

<img width="549" alt="Screen Shot 2020-11-12 at 12 01 03 AM" src="https://user-images.githubusercontent.com/5354757/98911867-29c02380-247a-11eb-9ff7-7606a0a48a07.png">

honestly, the difference in how long the individual strings are might warrant a whole separate layout for the starforged oracle list. that, or devising some more concise names for them 🤔

**Comment by raspilicious** [comment]
Have a look at how [RPGBot](https://top.gg/bot/rpgbot) handles shop inventory. It has multiple pages that you can switch between using navigation emoji reactions. This means that one post is made in the Discord channel that has its content replaced as you use the emojis to flip between each "page". This would work quite well with the Oracle tables, and would allow you to have a single column layout suitable for mobile as you were discussing above.

**Comment by XenotropicDev** [comment]
Fixed in commit 27c841f .
If this needs more adjustment a new issue will be created
